### PR TITLE
GRIDEDIT-801: Bind mkernel_mesh2d_convert_projection

### PR DESCRIPTION
**Pull request by ahmad-el-sayed**

### CODE REVIEW

**Review by andreasbuykx — APPROVED** [review]

File: MeshKernelNET.sln [hunk 1]
```diff
@@ -1,15 +1,15 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
```
[the remark this answers is not in the document]
Comment (ahmad-el-sayed) [reply MeshKernelNET.sln:2]: I think this happened when I added a new source, PrejectionOptions.cs. This changes the solution and it seems that I am using a newer VS version. thoughts?

File: test/MeshKernelNETTest/Api/MeshKernelTest.cs [hunk 2]
```diff
@@ -1543,6 +1543,81 @@ public void Mesh2dComputeOrthogonalizationThroughApi()
...
+
+                    Assert.AreEqual(0, api.Mesh2dSet(id, mesh));
+
+                    // Get data of initial mesh
+                    Assert.AreEqual(0, api.Mesh2dGetData(id, out meshInitial));
+                    Assert.IsNotNull(meshInitial);
+
+                    {
+                        // Set the zone string
+                        const string zone = "+proj=utm +lat_1=0.5 +lat_2=2 +n=0.5 +zone=31";
```
[the remark this answers is not in the document]
Comment (ahmad-el-sayed) [reply test/MeshKernelNETTest/Api/MeshKernelTest.cs:1568]: The back-end checks the format of the string (in fact boost::geometry does). If it is not valid an error is thrown.

File: test/MeshKernelNETTest/Api/MeshKernelTest.cs [hunk 3]
```diff
@@ -1543,6 +1543,81 @@ public void Mesh2dComputeOrthogonalizationThroughApi()
...
+
+                        // Round trip, convert back to cartesian
+                        Assert.AreEqual(0, api.Mesh2dConvertProjection(id, ProjectionOptions.Cartesian, zone));
+                        Assert.AreEqual(0, api.GetProjection(id, ref projection));
+                        Assert.AreEqual(ProjectionOptions.Cartesian, (ProjectionOptions)projection);
+                    }
+
+                    // Get data of final mesh
+                    Assert.AreEqual(0, api.Mesh2dGetData(id, out meshFinal));
+                    Assert.IsNotNull(meshFinal);
```
[the remark this answers is not in the document]
Comment (ahmad-el-sayed) [reply test/MeshKernelNETTest/Api/MeshKernelTest.cs:1584]: Maybe some additional assertions  between the conversions? at least that the mesh has changed ?

File: test/MeshKernelNETTest/Api/MeshKernelTest.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ahmad-el-sayed) [reply test/MeshKernelNETTest/Api/MeshKernelTest.cs:1568]: Added another test anyway

**Pull request merged**